yjit/src/utils.rs

```diff
@@ -91,6 +91,37 @@ macro_rules! c_callable {
     (fn $f:ident $args:tt $body:block) => { fn $f $args $body };
 }
 
+// Convert a CRuby UTF-8-encoded RSTRING into a Rust string.
+// This should work fine on ASCII strings and anything else
+// that is considered legal UTF-8, including embedded nulls.
+fn rstring_to_utf8_string(v: VALUE) -> String {
+    let str_ptr = unsafe { rb_rstring_dup(v) as *mut u8};
+    let str_len = unsafe { rb_RSTRING_LEN(v) } as usize;
+    let str_vec:Vec<u8> = unsafe { Vec::from_raw_parts(str_ptr, str_len, str_len) };
+    String::from_utf8(str_vec).unwrap()
+}
```

Comment:
There's a function named `rust_str_to_ruby`.

Can we rename `rstring_to_utf8_string ` to `ruby_str_to_rust` for consistency?

May also want to assert that the internal encoding is either ASCII or UTF-8

Reply:
Also this usage of Vec::from_raw_parts is not correct. It fails the first requirement for safety in the docs: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.from_raw_parts

This is related to the discussion in `rb_rstring_dup`.  You want to allocate in Rust and copy from the Ruby string instead of using xmalloc. Allocator mismatch breaks deallocation when the String is dropped.

Reply:
Huh. So slice::from_raw_parts does *not* take ownership of the buffer, but Boxing the slice *does* copy its data? That's... unintuitive. I mean, it says it copies the slice. But given that a slice is a reference into other storage, I had assumed it copied the *slice*, not the memory pointed to by the slice.